### ISSUE TITLE
Add sort-imports member sorting into a separate option

### ISSUE THREAD

**Issue by remcohaszing** [eslint/eslint#11019]
**What rule do you want to change?**

`sort-imports`

**Does this change cause the rule to produce more or fewer warnings?**

Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

Currently, `sort-imports` does two checks. One is always on, the other is optional. I would like to only use the optional part.

One part is that it checks the order of import statements. This can’t be turned off. I would like to check this using [import/order](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md), which offers more advanced features.

The part of the rule I would like to use, is the sorting of the imported members.

My proposal is to split the distinct `memberSort` option into a separate rule: `sort-import-members`.

**Please provide some example code that this change will affect:**

```js
// Builtins
import { join, resolve } from 'path';

// External
import { map } from 'lodash';
```

**What does the rule currently do for this code?**

`sort-imports` will cause conflicts with `import/order`, but also make sure import members are sorted nicely.

**What will the rule do after it's changed?**

`sort-imports` should be disabled to prevent conflicts with `import/order`.

`sort-import-members` will ensure the imported members are sorted, as `sort-imports` does now.

**Comment by kaicataldo** [comment]
Given the pluggable nature of ESLint and the size of the plugin community, we can't always ensure that all rules work with plugins. I'm not convinced we need an entirely new rule to solve this problem.

I wonder if a better solve would be to deprecate the [`ignoreMemberSort` option](https://eslint.org/docs/rules/sort-imports#options) in favor of an option that allows the toggling of sorting import declarations and imported members individually. I'm imagining something like:

```json
{
    "sort-imports": ["error", {
        "sortDeclarations": true,
        "sortMembers": true,
    }]
}
```

If we decide to go down this route, we should do this in a backwards compatible way.

**Comment by remcohaszing** [comment]
That proposal seems just as good to me. :+1: 

For backwards compatibility, I suggest the following signature:

```js
{
  "sort-imports": ["error", {
    ignoreMemberSort: false,  // Already exists
    ignoreDeclarationSort: false,  //  Analogous to `ignoreMemberSort`. The default matches the current behavious.
    ...otherOptions
  }]
}
```

**Comment by nzakas** [comment]
The proposal to add an option to turn off declaration sort seems reasonable to me. :+1:

@remcohaszing are you will to submit a pull request for this proposal?

**Comment by remcohaszing** [comment]
I'll give it a try somewhere this week.

**Comment by Fandekasp** [comment]
Can someone look into the related PR ? There hasn't been any action taken in over a month, and I'm also very interested in having a rule to check ordered and sorted imports.
Thank you for your PR @remcohaszing :)

**Comment by nzakas** [comment]
Thanks for the ping, we'll take a look.

**Comment by nzakas** [comment]
@ilyavolodin oh right, duh! Thanks.

**Comment by remcohaszing** [comment]
Is any action required from me? I thought not, but it’s taking much longer to get merged than I expected.

**Comment by nzakas** [comment]
@remcohaszing thanks for the ping. Sorry, a lot of team members are taking time off around this time of year, so things can take a bit longer than usual. It looks like enough team members support this change so I'm marking as accepted. The next step is to review the PR.
